clang/tools/driver/CacheLauncherMode.cpp

```diff
@@ -62,13 +62,16 @@ clang::handleClangCacheInvocation(SmallVectorImpl<const char *> &Args,
     Args[0] = Saver.save(compilerPath).data();
 
   if (isSameProgram(clangCachePath, compilerPath)) {
+    if (const char *SessionId = ::getenv("CLANG_CACHE_BUILD_SESSION_ID")) {
+      Args.append(
+          {"-fdepscan=daemon", "-fdepscan-share-identifier", SessionId});
+    } else {
+      Args.push_back("-fdepscan=inline");
```

Comment:
Not really a change from this commit, but I'm curious why you're passing `=inline`, vs. `-fdepscan` or `-fdepscan=auto` (and letting the compiler figure out the best thing)?

Reply:
FTR, @akyrtzi and I chatted offline.
- Currently, `-fdepscan=daemon` is only useful if sharing scanning state. As a result, `-fdepscan=auto` is the same as `-fdepscan=inline`.
- There are FIXMEs for `-fdepscan=daemon` to use the same daemon even when NOT sharing scanning state. This would not be a semantic change: idea is to share an OnDiskCAS connection, but to have different scanning arenas in the process, and only share `CachingOnDiskFileSystem` and `DependencyScanningService` in the same places it would have been previously. Once that's done, it'd be nice to change `-fdepscan=auto` (the default) to mean `-fdepscan=daemon` if that turns out to be faster.
- Later, the daemon could grow to do other work (not just depscanning), as a central build server.